File: vacation/vacation/Vacation.go
package vacation

import (
	"gopkg.in/jeevatkm/go-model.v1"
	
	"gorm.io/gorm"
	"fmt"
	"vacation/external"
)

type Vacation struct {
	gorm.Model
	Id int `gorm:"primaryKey" json:"id" type:"int"`
	StartDate undefined `json:"startDate"`
	EndDate undefined `json:"endDate"`
	Reason string `json:"reason"`
	UserId string `json:"userId"`
	Days int `json:"days"`
	Status string `json:"status"`

}

func (self *Vacation) onPostPersist() (err error){
	vacationRegistered := NewVacationRegistered()
	model.Copy(vacationRegistered, self)

	Publish(vacationRegistered)

	// Get request from Vacation
	vacation := &external.Vacation{}
	resp := external.GetVacation(vacation.Id)
	fmt.Println(resp)

	return nil
}
func (self *Vacation) onPrePersist() (err error){ return nil }
func (self *Vacation) onPreUpdate() (err error){ return nil }
func (self *Vacation) onPostUpdate() (err error){ return nil }
func (self *Vacation) onPreRemove() (err error){ return nil }
func (self *Vacation) onPostRemove() (err error){ return nil }

func (self *Vacation) Cancel(){
	vacationCancelled := NewVacationCancelled()
	model.Copy(vacationCancelled, self)
	Publish(vacationCancelled)
}
func (self *Vacation) Approve(approveCommand *{namePascalCase}}Command){
	vacationApproved := NewVacationApproved()
	model.Copy(vacationApproved, self)
	Publish(vacationApproved)
}
func (self *Vacation) ConfirmUsed(){
	vacationUsed := NewVacationUsed()
	model.Copy(vacationUsed, self)
	Publish(vacationUsed)
}
func (self *Vacation) Update(){
	vacationRejected := NewVacationRejected()
	model.Copy(vacationRejected, self)
	Publish(vacationRejected)
}

File: vacation/src/main/java/human/resource/mgmt/domain/Vacation.java
package human.resource.mgmt.domain;

import human.resource.mgmt.VacationApplication;
import human.resource.mgmt.domain.VacationRegistered;
import java.util.Date;
import java.util.List;
import javax.persistence.*;
import lombok.Data;

@Entity
@Table(name = "Vacation_table")
@Data
public class Vacation {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String id;

    private Date startDate;

    private Date endDate;

    private String reason;

    private String userId;

    private Integer days;

    private String status;

    @PostPersist
    public void onPostPersist() {
        VacationRegistered vacationRegistered = new VacationRegistered(this);
        vacationRegistered.publishAfterCommit();
        // Get request from Vacation
        //human.resource.mgmt.external.Vacation vacation =
        //    Application.applicationContext.getBean(human.resource.mgmt.external.VacationService.class)
        //    .getVacation(/** mapping value needed */);

    }

    public static VacationRepository repository() {
        VacationRepository vacationRepository = VacationApplication.applicationContext.getBean(
            VacationRepository.class
        );
        return vacationRepository;
    }

    public void cancel(CancelCommand cancelCommand) {
        VacationCancelled vacationCancelled = new VacationCancelled(this);
        vacationCancelled.publishAfterCommit();
    }

    public void approve(ApproveCommand approveCommand) {
        VacationApproved vacationApproved = new VacationApproved(this);
        vacationApproved.publishAfterCommit();
    }

    public void confirmUsed(ConfirmUsedCommand confirmUsedCommand) {
        VacationUsed vacationUsed = new VacationUsed(this);
        vacationUsed.publishAfterCommit();
    }

    public void update(UpdateCommand updateCommand) {
        VacationRejected vacationRejected = new VacationRejected(this);
        vacationRejected.publishAfterCommit();
    }
}
